### ISSUE TITLE
Keyboard focus border right part is not visible in contextual menu item.

### ISSUE THREAD

**Issue by KeiraYang** [microsoft/fluentui#14795]
Repro steps:
1. Visit https://developer.microsoft.com/en-us/fluentui#/controls/web/contextualmenu
2. Use keyboard to open the contextual menu example.
3. Use keyboard to access each item. Observe the issue. The focus border right part is not visible.

![image](https://user-images.githubusercontent.com/28148759/91523636-d97ffd80-e92f-11ea-9e58-3afa04b0957b.png)

**Comment by paulgildea** [comment]
@KeiraYang Thanks for the feedback!

Could you provide what platform and browser you are seeing the issue in? I'd like to understand if this is a recent regression - Thanks!

I'm using:
Edge - Version 84.0.522.63 (Official build) (64-bit)
macOS Catalina 10.15.4
And this what I see... 😕 
![image](https://user-images.githubusercontent.com/33229964/91587097-269da880-e90b-11ea-961b-41139e00880f.png)

**Comment by KeiraYang** [comment]
Hi @paulgildea . I found the issue when using Windows + Edge Chromium. Thanks to take a look!

**Comment by KeiraYang** [comment]
Hi @paulgildea @khmakoto Could you please take a look at this issue? Thanks!

**Comment by dev-a11y** [comment]
Hi, I also experience this issue (Chrome 85.x on Windows) - looks like sub-pixel rendering, take a look at below:
![image](https://user-images.githubusercontent.com/59690753/93489322-a9f95b00-f907-11ea-9c5e-735ace672c7a.png)

It may be worth to do the change in positioning logic so that Callout position doesn't have fractional pixels. On the other hand sub-pixel rendering may also happen for controls that doesn't display in Callouts/Panels/Modals BUT in those places developers usually have control over wrappers - here positioning logic is calculating position so probably no way to workaround in client code? One thing I'm not sure though is if such fix won't hide more fundamental issue - probably not and such fix would be enough...

One thing to mention here is that the issue is not related to keyboard focus border that was added quite recently - I see this in the earlier fabric versions too but the issue wasn't so exposed before as it impacted grey highlight which wasn't so easy to notice at first glance:
![image](https://user-images.githubusercontent.com/59690753/93492027-a74c3500-f90a-11ea-9b4b-e2908e7e2a3f.png)

**Comment by dev-a11y** [comment]
Hi @xugao, @dzearing could you please take a look at this issue?

**Comment by dzearing** [comment]
@joschect is the positioning logic landing on fractional values? We should probably floor them, if so.

**Comment by joschect** [comment]
Right now for accuracy it does land on fractional values. We can just use floor. I had thought that the browser rounded pixel fractional placements but I haven't found much indication that that is the case. I did a bit of research in this [codepen](https://codepen.io/slimemold/pen/MWyZMPW?editors=0110) that suggests that for absolute positioning, that the fractionals are rounded and placement is based on how the browser rounds. OTOH it appears that the _content_ positions itself based, at least somewhat, on the fraction position. It makes sense to just floor it moving forward I think.

**Comment by paulgildea** [comment]
@joschect Sounds like we have a root cause and a solution.

@KeiraYang / @dev-a11y Interested in making a PR? 😊

**Comment by dev-a11y** [comment]
Me not - please go ahead. I would need to set up workspace/clone repo/install deps etc just for this fix - don't want to
Below line may require mentioned flooring but not 100% sure..

https://github.com/microsoft/fluentui/blob/80bf8cc7951f154a975e8b0f4bb082be658cf167/packages/office-ui-fabric-react/src/utilities/positioning/positioning.ts#L123

Hope this will be released soon ... currently those dropdowns, context menus and list pickers may look quite sloppy (not on every monitor and resolution but still) :confused:

**Comment by dev-a11y** [comment]
I have changed below function (positioning.js file in node_modules\office-ui-fabric-react\lib\utilities\positioning):
```javascript
function _getRelativeEdgeDifference(rect, hostRect, edge) {
    var edgeDifference = _getEdgeValue(rect, edge) - _getEdgeValue(hostRect, edge);
    return _getRelativeEdgeValue(edge, edgeDifference);
}
```
into:
```javascript
function _getRelativeEdgeDifference(rect, hostRect, edge) {
    var edgeDifference = _getEdgeValue(rect, edge) - _getEdgeValue(hostRect, edge);
    var relativeEdgeValue = _getRelativeEdgeValue(edge, edgeDifference);
    if (edge === RectangleEdge.left) {
        return Math.ceil(relativeEdgeValue);
    }
    if (edge === RectangleEdge.right) {
        return Math.floor(relativeEdgeValue);
    }
    return relativeEdgeValue;
}
```
and the issue is no longer reproducible in any of the mentioned components. I was thinking whether to add opposite conditions for RTL or not but ultimately found it not necessary. Could you please implement such change to that file @joschect? And your hypothesis that browsers have problems with content positioning even though rounding seems to be working for absolute positioned parents is highly correct imho - especially for flex positioned content I would add. Here is [an interesting read](https://chenhuijing.com/blog/about-subpixel-rendering-in-browsers/) among many that gives some insights on related topics (but unfortunately putting _thin_ instead _1px_ as suggested there is not solving mentioned here problem).
